migrations/migration-9/pin-set.js

```diff
@@ -212,22 +219,30 @@ function storeItems (blockstore, items) {
         idx++
       }
 
-      return new DAGNode(headerBuf, fanoutLinks)
+      return {
+        Data: headerBuf,
+        Links: fanoutLinks
+      }
     }
 
     /**
-     * @param {DAGNode} child
+     * @param {PBNode} child
      * @param {number} binIdx
      */
     async function storeChild (child, binIdx) {
-      const buf = dagpb.util.serialize(child)
-      const cid = await dagpb.util.cid(buf, {
-        cidVersion: 0,
-        hashAlg: multihash.names['sha2-256']
-      })
+      const buf = dagPb.encode(child)
+      const digest = await sha256.digest(buf)
+      const cid = CID.createV0(digest)
+
       await blockstore.put(cidToKey(cid), buf)
 
-      fanoutLinks[binIdx] = new DAGLink('', child.size, cid)
+      let size = child.Links.reduce((acc, curr) => acc + (curr?.Tsize || 0), 0) + buf.length
```

Comment:
repetitive pattern suggests we might need to pull up a utility or two to dagPb to deal with `Tsize`

Reply:
and, is `curr?.Tsize` safe to use these days in the platforms we care about? I haven't even bothered touching it, assuming it'll be a year or two away for general use

Reply:
> repetitive pattern suggests

Yes, this was previously handled by the `ipld-dag-pb` module, but that feature wasn't ported over so a utility would be helpful.

> and, is `curr?.Tsize` safe to use these days in the platforms we care about?

I thought yes, but it looks like the version of acorn used by the current `create-react-app` release doesn't support it.  A later version does but it'll require them to upgrade to webpack 5 and who knows when that will land.

It's caused a few problems with the last js-ipfs release so we should probably back out our use of optional chaining for the time being.